js/main.js

```diff
@@ -0,0 +1,206 @@
+
+      const $trailer = document.createElement('iframe');
+      $trailer.setAttribute('class', 'trailer');
+      $trailer.setAttribute('src', top25Data[i].trailer.embed_url);
+      $trailer.setAttribute('frameborder', '0');
+      $trailer.setAttribute('allow', 'accelerometer, clipboard-write; encrypted-media; gyroscope; picture-in-picture'
+      );
+      $trailer.setAttribute('allowfullscreen', 'allowfullscreen');
+      $trailerDiv.appendChild($trailer);
+
+      const $watchNowDiv = document.createElement('div');
+      $watchNowDiv.setAttribute('class', 'watch-now-wrap');
+      $modalContainer.appendChild($watchNowDiv);
+
+      const $watchNow = document.createElement('a');
+      $watchNow.textContent = 'Watch Now';
+      const websiteLink = top25Data[i].url;
+      $watchNow.setAttribute('href', websiteLink);
+      $watchNow.setAttribute('class', 'watch-now');
+      $watchNowDiv.appendChild($watchNow);
+
+      const $modalBtnWrap = document.createElement('div');
+      $modalBtnWrap.setAttribute('class', 'modal-btn-wrap');
+      $modalContainer.appendChild($modalBtnWrap);
+
+      const $modalAddBtn = document.createElement('button');
+      $modalAddBtn.textContent = '+ Add';
+      $modalAddBtn.setAttribute('class', 'add modal-btn');
+      $modalBtnWrap.appendChild($modalAddBtn);
+
+      const $modalCloseBtn = document.createElement('button');
+      $modalCloseBtn.textContent = 'Close';
+      $modalCloseBtn.setAttribute('class', 'close modal-btn');
+      $modalBtnWrap.appendChild($modalCloseBtn);
+      // end of modal elements
+    }
+
+    // modal event listeners
+    const $defaultModal = document.querySelector('.main-container');
+    const $modal = document.querySelectorAll('.modal');
+    const $ol = document.querySelector('#anime-list');
+    const $closeBtnClicked = document.querySelectorAll('.close');
+
+    // modal pop up if user clicks on a show (added a conditional bc clicking outside the list showed error, since the event listener was on the ol and buttons in the modal are still part of the ol)
+    $ol.addEventListener('click', function openModal(event) {
+      if (event.target.tagName !== 'OL' && event.target.tagName !== 'LI' && event.target.tagName !== 'BUTTON') {
+        const $parentElement = event.target.closest('.list-wrap');
+        const $closestModal = $parentElement.nextElementSibling;
+        $defaultModal.className = 'main-container modal-opened';
+        $closestModal.className = 'show modal';
+      }
+    });
+
+    // close modal button (needed to use loop because close button event needs to be applied for all 25 list items)
+    for (let i = 0; i < $closeBtnClicked.length; i++) {
+      $closeBtnClicked[i].addEventListener('click', function closeModal(event) {
+        $defaultModal.className = 'main-container';
+        $modal[i].className = 'hidden modal';
+      });
+    // end of modal event listeners
+    }
+  });
+  xhr.send();
+}
+
+getAnimeData();
```

Comment:
I would call this inside of an event listener instead. Like `DOMContentLoaded`

Reply:
put `xhr.send()` and `getAnimeData` function  inside `window event listener` with type `DOMContentLoaded`